packages/nx/src/command-line/migrate.ts

```diff
@@ -177,7 +182,7 @@ export class Migrator {
       }
     }
 
-    const packages = this.collapsePackages(
+    const packages = await this.collapsePackages(
       targetPackage,
       targetVersion,
       migrationsJson
```

Comment:
I noticed that after I created the PR and I was still looking into possible solutions. I forgot to put the PR in Draft.

The first choice does worsen the perf a lot, I tried it. The second one would still impact perf because we need to defer some collection until after the prompts are answered (the decision impacts what we collect). I'll spend a bit more trying to see if there's a way to avoid the funky rendering without removing the concurrent calls.